### PR TITLE
rpc net module

### PR DESCRIPTION
**Pull request by lzl124631x**
* added support for net module

This `net_version` API is needed for MM.

### CODE REVIEW

**Comment by mikedoan** [comment]
i'm guessing
networkID is like ethereum ropsten, or rinkeby 
networkVersion is the release version?

Can you elaborate that? in general, lgtm

**Comment by lzl124631x** [comment]
It's the ID of different testnet. Each testnet has a name (like ropsten, or rinkeby) and ID (number).

**Comment by lzl124631x** [comment]
Jenkins start build

**Comment by lzl124631x** [comment]
Jenkins, test this please.

**Comment by lzl124631x** [comment]
Jenkins, please test this.

**Comment by lzl124631x** [comment]
Jenkins, test this please.

**Comment by lzl124631x** [comment]
Jenkins, test this please.

**Comment by lzl124631x** [comment]
Jenkins, test this please.

**Comment by lzl124631x** [comment]
Jenkins, test this please.

File: internal/hmyapi/net.go [hunk 1]
```diff
@@ -9,21 +9,21 @@ import (
...
 
 // PeerCount returns the number of connected peers
 func (s *PublicNetAPI) PeerCount() hexutil.Uint {
 	return hexutil.Uint(s.net.GetPeerCount())
 }
 
-// NetworkID ...
-func (s *PublicNetAPI) NetworkID() string {
-	return fmt.Sprintf("%d", 1) // TODO(ricl): we should add support for network id (https://github.com/ethereum/wiki/wiki/JSON-RPC#net_version)
+// Version ...
```
Comment (mikedoan) [inline internal/hmyapi/net.go]: write more comment

File: hmy/backend.go [hunk 2]
```diff
@@ -55,6 +57,7 @@ func New(nodeAPI NodeAPI, txPool *core.TxPool, eventMux *event.TypeMux) (*Harmon
 		chainDb:        chainDb,
 		bloomIndexer:   NewBloomIndexer(chainDb, params.BloomBitsBlocks, params.BloomConfirms),
 		nodeAPI:        nodeAPI,
+		networkID:      1, // TODO(ricl): this should be from config
```
Comment (mikedoan) [inline hmy/backend.go]: There is a networkVersion below and here networkID.

Those 2 terms make me very confused. Can you write comments as your clarification?

Comment (lzl124631x) [reply hmy/backend.go]: agreed it's confusing. For now it's for sticking to eth's standard. I updated the comments.

File: internal/hmyapi/net.go [hunk 3]
```diff
@@ -9,21 +9,21 @@ import (
...
 // PeerCount returns the number of connected peers
 func (s *PublicNetAPI) PeerCount() hexutil.Uint {
 	return hexutil.Uint(s.net.GetPeerCount())
 }
 
-// NetworkID ...
-func (s *PublicNetAPI) NetworkID() string {
-	return fmt.Sprintf("%d", 1) // TODO(ricl): we should add support for network id (https://github.com/ethereum/wiki/wiki/JSON-RPC#net_version)
+// Version returns the network version, i.e. network ID identifying which network we are using
+func (s *PublicNetAPI) Version() string {
```
Comment (mikedoan) [inline internal/hmyapi/net.go]: Version here and NetVersion above.

**Comment by LeoHChen** [comment]
@lzl124631x , why do we need to networkID?  We have no networkID concept in Harmony.  For API compatibility purpose?  Is Metamask using it?

File: internal/hmyapi/net.go (hunk 3 above)
Comment (lzl124631x) [reply internal/hmyapi/net.go]: Have to make it name "Version" because the API is `net_version`.

**Comment by lzl124631x** [comment]
@LeoHChen 
1. going forward I think we also need networkID. For example, we have networkID = 1 for public testnet and networkID = 2 for private testnet version 1, networkID = 3 for nightly build etc. This allows different versions of network coexists.
2. Metamask is using it when we switch between networks (like MainNet and Ropsten)

image

**Pull request merged**